### PR TITLE
cppcommon: Bump to 1.0.1.0

### PR DESCRIPTION
**Pull request by alejandro-colomar**
**cppcommon/1.0.1.0**

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by alejandro-colomar** [comment]
@tarc , @prince-chrismc , could you please have a look at the failure and help me?  You may know what to do, as you packaged version 1.0.0.0

**Comment by prince-chrismc** [comment]
bug in FMT

this project "lives" on master

the fix has not been released yet https://github.com/fmtlib/fmt/commit/ac352081159f0aebe707fc55e216984f75c7e6ad

**Comment by alejandro-colomar** [comment]
D'oh!  Do you think I could patch the conan fmt/7.1.3 package to fix this?

**Comment by alejandro-colomar** [comment]
I'll close this temporarily until fmt releases a new version.

**Comment by alejandro-colomar** [comment]
I see that fmt/8.0.0 has been released recently, and also packaged.

I'm going to reopen this PR.

Will I make worse the CI problems of today if I do it now?

**Comment by alejandro-colomar** [comment]
I rebased, as there are no reviews yet, and it's been a long time since those commits.  Just to be sure.

File: recipes/cppcommon/all/patches/0001-update-cmakelists-1-0-1-0.patch [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+   list(APPEND LINKLIBS ${LIBRT_LIBRARIES})
+-  list(APPEND LINKLIBS ${LIBUUID_LIBRARIES})
++  list(APPEND LINKLIBS CONAN_PKG::libuuid)
+ endif()
+ if(WIN32 OR MSYS)
+   list(APPEND LINKLIBS ${DBGHELP_LIBRARIES})
+   list(APPEND LINKLIBS ${RPC_LIBRARIES})
+-  list(APPEND LINKLIBS ${USERENV_LIBRARIES})
++  list(APPEND LINKLIBS userenv)
++  list(APPEND LINKLIBS rpcrt4)
```
Comment (prince-chrismc) [inline recipes/cppcommon/all/patches/0001-update-cmakelists-1-0-1-0.patch:59]: Is the find `rpc` not getting it?

Comment (alejandro-colomar) [reply recipes/cppcommon/all/patches/0001-update-cmakelists-1-0-1-0.patch:59]: That is a copy&paste of the patch for cppcommon/1.0.0.0.  See <https://github.com/conan-io/conan-center-index/blob/08f8662c2705aed9235ade59cd22b01336ef7f1d/recipes/cppcommon/all/patches/0001-update-cmakelists-1-0-0-0.patch>.

I don't know why it was patched.  I still don't know too much of CMake...

BTW, for the future, having useful commit messages for those patches is a good idea :-)

Comment (prince-chrismc) [reply recipes/cppcommon/all/patches/0001-update-cmakelists-1-0-1-0.patch:59]: > BTW, for the future, having useful commit messages for those patches is a good idea

Do you know any tools which can check for that? I wish this was something we could enforce

**Review by prince-chrismc — APPROVED** [review]

File: recipes/cppcommon/all/patches/0001-update-cmakelists-1-0-1-0.patch (hunk 1 above)
Comment (alejandro-colomar) [reply recipes/cppcommon/all/patches/0001-update-cmakelists-1-0-1-0.patch:59]: I think you could use git hooks to check if any new files with the extension .patch are added.

If so, you could use checkpatch (I haven't used it for this, but for static analysis of .c source files, but it also checks some things about git patches; I'll check) to check that the patches are git patches, and not just diffs.

I never did that, so let me check.  I'll come back with something.

Comment (alejandro-colomar) [reply recipes/cppcommon/all/patches/0001-update-cmakelists-1-0-1-0.patch:59]: Humm, checkpatch complains about patches that don't have a valid 'Signed-off-by: Name <email>' line.

But it doesn't complain about any other metadata of patches.
And the worst part is that that check is only triggered if the patch is a git patch;
if it's just a diff, it doesn't complain, and will only check the source of the patch.

So maybe a simple bash/python script that checks the contents of the patch would serve this purpose.
It should find a line that begins with `/^Subject: /` before it finds a line that begins with `/^diff/`.

The following should be enough:
```bash
alx@lvl01026:~/src/conan/conan-center-index/recipes/cppcommon/all/patches$ cat check_patch_format.sh 
#!/bin/bash
set -Eeuo pipefail;

EX_OK=0;
EX_DATAERR=65;

patch="$1";

< "$patch" \
sed '/^diff/,$d' \
| grep -q '^Subject: ' \
&& exit ${EX_OK};

1>&2 echo "$patch is not a git patch.";
1>&2 echo "Please use git-format-patch(1) to create the patch.";
exit ${EX_DATAERR};

alx@lvl01026:~/src/conan/conan-center-index/recipes/cppcommon/all/patches$ ./check_patch_format.sh 0001-update-cmakelists-1-0-0-0.patch 
0001-update-cmakelists-1-0-0-0.patch is not a git patch.
Please use git-format-patch(1) to create the patch.
alx@lvl01026:~/src/conan/conan-center-index/recipes/cppcommon/all/patches$ echo $?
65
alx@lvl01026:~/src/conan/conan-center-index/recipes/cppcommon/all/patches$ ./check_patch_format.sh 0001-update-cmakelists-1-0-1-0.patch 
alx@lvl01026:~/src/conan/conan-center-index/recipes/cppcommon/all/patches$ echo $?
0

```

Comment (alejandro-colomar) [reply recipes/cppcommon/all/patches/0001-update-cmakelists-1-0-1-0.patch:59]: In the meantime, I'll report that behavior to the maintainers of checkpatch, to see if they want to fix it.

**Comment by alejandro-colomar** [comment]
Are the problems with the CI server still present?  This job has been running for around 2 days :/

File: recipes/cppcommon/all/patches/0001-update-cmakelists-1-0-1-0.patch (hunk 1 above)
Comment (prince-chrismc) [reply recipes/cppcommon/all/patches/0001-update-cmakelists-1-0-1-0.patch:59]: Thanks for the idea!

**Review by prince-chrismc — APPROVED** [review]

File: recipes/cppcommon/all/patches/0001-update-cmakelists-1-0-1-0.patch (hunk 1 above)
Comment (jgsogo) [reply recipes/cppcommon/all/patches/0001-update-cmakelists-1-0-1-0.patch:59]: We can enforce some extra fields in the `conandata.yml` for the patches (using a hook). We are upgrading Conan in the CI and now it won't fail if there are some extra fields. I'm talking about this kind of information: https://github.com/jgsogo/conan-center-index/pull/4/files

**Review by jgsogo — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**